viewing_party/party.py

```diff
@@ -0,0 +1,186 @@
+    avg_rating = 0
+    watched = user_data["watched"]
+    if len(watched) == 0:
+        return 0.0
+    else:
+        for movie in watched:
+            avg_rating += movie["rating"]
+        return avg_rating/len(watched)
+
+def get_most_watched_genre(user_data):
+    genre_dict = {}
+    most_watched = 0
+    most_watched_genre = " "
+    watched = user_data["watched"]
+    if len(watched) == 0:
+        return None
+    for movie in watched:
+        if movie["genre"] not in genre_dict:
+            genre_dict[movie["genre"]] = 1
+        else: 
+            genre_dict[movie["genre"]] += 1
+    for genre, count in genre_dict.items():
+        if count > most_watched:
+            most_watched = count
+            most_watched_genre = genre
+    return most_watched_genre
+
+# Wave 3
+# This function consolidates friends' watched list values into one list. 
+def get_movie_list_from_watched_dict(list):
+    all_movies_watched_by_friends = []
+    for friend_movie_list in list:
+        for i in range(len(friend_movie_list["watched"])):
+            popped_movie = friend_movie_list["watched"].pop()
+            all_movies_watched_by_friends.append(popped_movie)
+    return all_movies_watched_by_friends
+
+def get_unique_watched(user_data):
+    unique_movies = []
+    watched = user_data["watched"]
+    friends_list = user_data["friends"]
+    friends_movie_list = get_movie_list_from_watched_dict(friends_list)
+    for movie in watched:
+        if movie in friends_movie_list:
+            continue
+        else:
+            unique_movies.append(movie)
+    return unique_movies
+
+def get_friends_unique_watched(user_data):
+    friends_unique_movies = []
+    watched = user_data["watched"]
+    friends_list = user_data["friends"]
+    friends_movie_list = get_movie_list_from_watched_dict(friends_list)
+    for movie in friends_movie_list:
+        if movie in watched or movie in friends_unique_movies:
+            continue 
+        else:
+            friends_unique_movies.append(movie)
+    return friends_unique_movies
+
+# Wave 4
+# This function consolidates friends' watched and host values into one list.
+def get_friends_movie_info(user_data):
+    all_friends_movie_info = []
+    friends = user_data["friends"]
+    for friend in friends:
+        for i in range(len(friends)):
+            if len(friend["watched"]) > 0:
+                friend_movie_info = friend["watched"].pop()
+                all_friends_movie_info.append(friend_movie_info)
+            else:
+                continue
+    return all_friends_movie_info
+
+# This function creates a list of movie titles from the user's watched list.
+def get_user_movie_titles(user_data):
+    user_movie_titles = []
+    watched = user_data["watched"]
+    if len(watched) > 0:
+        for i in range(len(watched)):
+            user_movie_titles.append(watched[i]["title"])
+    print(user_movie_titles)
+    return user_movie_titles
+
+
+def get_available_recs(user_data):
```

Comment:
great that you used helper functions here. How could you use the functions already in the project, like `get_friends_unique_watched(user_data)`?